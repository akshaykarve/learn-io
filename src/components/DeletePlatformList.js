import React from 'react';
import Platform from './Platform.js';
import './ComponentStyle.css'
import deleteIcon from './images/delete.png'
const DeletePlatformList=({platforms, onChangeDelete})=>{
	return(
		// loop for all platforms
		<div >
  			{
  				platforms.map((user,i)=>{
					return (
						<div className='deleteList' key={i} >
							<Platform name={platforms[i].platformName}/>
							<button onClick={()=>onChangeDelete(platforms[i],true)} className='deleteButton'><img src={deleteIcon} height='40px' width='40px' alt="delete"/></button>
						</div>
					);
				})
			}
  		</div>
	);
}
export default DeletePlatformList;