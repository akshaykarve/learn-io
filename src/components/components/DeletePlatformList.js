import React from 'react';
import '../ComponentStyle.css';
import Platform from './Platform';
import deleteIcon from '../images/delete.png';

const DeletePlatformList=({platforms, setDeletePlatform,setSelectPlatform})=>{
	return(
		// loop for all platforms
		<div >
  			{
  				platforms.map((user,i)=>{
					return (
						<div className='deleteList' key={i} >
							<Platform name={platforms[i].platformName} platform={platforms[i]} setSelectPlatform={setSelectPlatform}/>
							<button onClick={()=>setDeletePlatform(platforms[i])} className='deleteButton'><img src={deleteIcon} height='40px' width='40px' alt="delete"/></button>
						</div>
					);
				})
			}
  		</div>
	);
}

export default DeletePlatformList;