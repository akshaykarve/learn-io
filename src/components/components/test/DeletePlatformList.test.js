import React from 'react';
import renderer from 'react-test-renderer';
import {default as DeletePlatformList} from "../DeletePlatformList";

describe("Delete Platform List Component", ()=>{
	it('test three platforms', () => {
		
		const tree = renderer.create(
			<DeletePlatformList platforms={[{platformName:"test"},{platformName:"test2"},{platformName:"test3"}]}
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('test no platform', () => {
		
		const tree = renderer.create(
			<DeletePlatformList platforms={[]}
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('test four platforms', () => {
		
		const tree = renderer.create(
			<DeletePlatformList platforms={[{platformName:"test"},{platformName:"test2"},{platformName:"test3"},{platformName:"test4"}]}
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});
});