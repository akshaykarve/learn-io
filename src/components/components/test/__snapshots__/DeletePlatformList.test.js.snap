// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete Platform List Component test four platforms 1`] = `
<div>
  <div
    className="deleteList"
  >
    <div
      className="platformStyle grow"
      onClick={[Function]}
    >
      <img
        alt="platformImage"
        src="https://robohash.org/test?200x200"
      />
      <div>
        <h4>
          test
        </h4>
      </div>
    </div>
    <button
      className="deleteButton"
      onClick={[Function]}
    >
      <img
        alt="delete"
        height="40px"
        src="delete.png"
        width="40px"
      />
    </button>
  </div>
  <div
    className="deleteList"
  >
    <div
      className="platformStyle grow"
      onClick={[Function]}
    >
      <img
        alt="platformImage"
        src="https://robohash.org/test2?200x200"
      />
      <div>
        <h4>
          test2
        </h4>
      </div>
    </div>
    <button
      className="deleteButton"
      onClick={[Function]}
    >
      <img
        alt="delete"
        height="40px"
        src="delete.png"
        width="40px"
      />
    </button>
  </div>
  <div
    className="deleteList"
  >
    <div
      className="platformStyle grow"
      onClick={[Function]}
    >
      <img
        alt="platformImage"
        src="https://robohash.org/test3?200x200"
      />
      <div>
        <h4>
          test3
        </h4>
      </div>
    </div>
    <button
      className="deleteButton"
      onClick={[Function]}
    >
      <img
        alt="delete"
        height="40px"
        src="delete.png"
        width="40px"
      />
    </button>
  </div>
  <div
    className="deleteList"
  >
    <div
      className="platformStyle grow"
      onClick={[Function]}
    >
      <img
        alt="platformImage"
        src="https://robohash.org/test4?200x200"
      />
      <div>
        <h4>
          test4
        </h4>
      </div>
    </div>
    <button
      className="deleteButton"
      onClick={[Function]}
    >
      <img
        alt="delete"
        height="40px"
        src="delete.png"
        width="40px"
      />
    </button>
  </div>
</div>
`;

exports[`Delete Platform List Component test no platform 1`] = `<div />`;

exports[`Delete Platform List Component test three platforms 1`] = `
<div>
  <div
    className="deleteList"
  >
    <div
      className="platformStyle grow"
      onClick={[Function]}
    >
      <img
        alt="platformImage"
        src="https://robohash.org/test?200x200"
      />
      <div>
        <h4>
          test
        </h4>
      </div>
    </div>
    <button
      className="deleteButton"
      onClick={[Function]}
    >
      <img
        alt="delete"
        height="40px"
        src="delete.png"
        width="40px"
      />
    </button>
  </div>
  <div
    className="deleteList"
  >
    <div
      className="platformStyle grow"
      onClick={[Function]}
    >
      <img
        alt="platformImage"
        src="https://robohash.org/test2?200x200"
      />
      <div>
        <h4>
          test2
        </h4>
      </div>
    </div>
    <button
      className="deleteButton"
      onClick={[Function]}
    >
      <img
        alt="delete"
        height="40px"
        src="delete.png"
        width="40px"
      />
    </button>
  </div>
  <div
    className="deleteList"
  >
    <div
      className="platformStyle grow"
      onClick={[Function]}
    >
      <img
        alt="platformImage"
        src="https://robohash.org/test3?200x200"
      />
      <div>
        <h4>
          test3
        </h4>
      </div>
    </div>
    <button
      className="deleteButton"
      onClick={[Function]}
    >
      <img
        alt="delete"
        height="40px"
        src="delete.png"
        width="40px"
      />
    </button>
  </div>
</div>
`;
